### ISSUE TITLE
Poco package only uses "Basic Edition"

### ISSUE THREAD

**Issue by Krzmbrzl** [mxe/mxe#2552]
From [their website](https://pocoproject.org/download.html#packages):
> The Basic Edition only contains the Foundation, JSON, XML, Util and Net libraries, but does not require any external dependencies.

Thus all other components are not built (for my case in particular Zip is the essential missing one).

This happens because the version is set to
https://github.com/mxe/mxe/blob/228cf95a697aa838c79bd1ec6d9cf4c434e83636/src/poco.mk#L7

Thus the script will try to download `poco-1.9.4.tar.gz` which sadly is the Basic Edition of Poco. The complete Poco package is named `poco-1.9.4-all.tar.gz` (Note the `-all` postfix)

I am currently testing whether simply adding `-all` to the version is enough to build all of Poco. If so, I'll submit a PR to fix this.

**Comment by Krzmbrzl** [comment]
Okay it seems as if it isn't quite that easy:
```
Failed to build package poco for target i686-w64-mingw32.static!
------------------------------------------------------------
Makefile:301: recipe for target 'PageCompiler-libexec' failed
make[2]: *** [PageCompiler-libexec] Error 2
make[2]: Leaving directory '/home/robert/Documents/Git/mxe/tmp-poco-i686-w64-mingw32.static/poco-1.9.4-all'
Makefile:827: recipe for target 'build-only-poco_i686-w64-mingw32.static' failed
make[1]: *** [build-only-poco_i686-w64-mingw32.static] Error 2
make[1]: Leaving directory '/home/robert/Documents/Git/mxe'
real	6m50,870s
user	10m59,838s
sys	2m26,648s
```

I'll have to look into this some more

**Comment by davidebeatrici** [comment]
We have our own entry point called `MumbleMain()`, which is defined and exported when we build as static (loader + DLL).

About `WinMain()`: https://docs.microsoft.com/en-us/windows/win32/learnwin32/winmain--the-application-entry-point

The Makefile for the target is here: https://github.com/pocoproject/poco/blob/master/PageCompiler/Makefile

This is the configuration used by MXE: https://github.com/pocoproject/poco/blob/master/build/config/MinGW-CrossEnv

For now I didn't find anything that tells MinGW to search for `WinMain()`. It's possible it doesn't find `main()` and falls back to the other entry point.

I would try to build with CMake.

**Comment by Krzmbrzl** [comment]
Yeah that might be worth a try. I'll give it a shot tomorrow :+1:

**Comment by Krzmbrzl** [comment]
Turns out that the patches that enable cmake build on MinGW are still in the `development-experimental` branch of Poco. And since I don't want to start patching a lot (and building off an experimental branch is not a good option either) I stayed with autoconf. I simply disabled the PageCompiler since it seems that's not a library anyways :shrug:
